### PR TITLE
Users/kundadebdatta/4364 vector index msdata direct

### PR DESCRIPTION
**Pull request by kundadebdatta**
# Pull Request Template

## Description

High level changes for introducing vector similarity in .NET SDK. This is just a sample PR and should not be merged.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ContainerProperties.cs [hunk 1]
```diff
@@ -289,6 +292,26 @@ public IndexingPolicy IndexingPolicy
             }
         }
 
+        /// <summary>
+        /// Gets or sets the client encryption policy information for storing items in a container from the Azure Cosmos service.
```
Comment (Pilchie) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/ContainerProperties.cs:296]:
```suggestion
        /// Gets or sets the vector embedding policy information for storing items in a container from the Azure Cosmos service.
```

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ContainerProperties.cs [hunk 2]
```diff
@@ -289,6 +292,26 @@ public IndexingPolicy IndexingPolicy
...
+        /// </summary>
+        /// <value>
+        /// It is an optional property.
+        /// By default, ClientEncryptionPolicy is set to null meaning the feature is turned off for the container.
```
Comment (Pilchie) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/ContainerProperties.cs:300]:
```suggestion
        /// By default, VectorEmbeddingPolicy is set to null meaning the feature is turned off for the container.
```

File: Microsoft.Azure.Cosmos/src/Resource/Settings/DistanceFunction.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    /// <summary>
+    /// Defines the distance function for a vector index specification in the Azure Cosmos DB service.
+    /// </summary>
+    public enum DistanceFunction
+    {
+        /// <summary>
+        /// Represents the euclidean distance function.
+        /// </summary>
+        [EnumMember(Value = "Euclidean")]
```
Comment (Pilchie) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/DistanceFunction.cs:16]: Can you use `nameof` here?

```suggestion
        [EnumMember(Value = nameof(Euclidean))]
```

File: Microsoft.Azure.Cosmos/src/Resource/Settings/VectorEmbeddingPolicy.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            if (path[0] != '/' || path.LastIndexOf('/') != 0)
+            {
+                throw new ArgumentException("The argument {0} is not a valid path.", path);
+            }
+        }
+
+        private static void ValidateEmbeddingDimensions(
+            long dimensions)
+        {
+            if (dimensions == null || dimensions < 1)
```
Comment (Pilchie) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/VectorEmbeddingPolicy.cs:78]: `dimensions` is a long, and therefore can't be null.

```suggestion
            if (dimensions < 1)
```

File: Microsoft.Azure.Cosmos/src/Resource/Settings/VectorIndexPath.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+    /// <summary>
+    /// Defines the target data type of an vector index path specification in the Azure Cosmos DB service.
+    /// </summary>
+    public enum VectorIndexType
+    {
+        /// <summary>
+        /// Represents a flat vector index type.
+        /// </summary>
+        [EnumMember(Value = "Flat")]
```
Comment (Pilchie) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/VectorIndexPath.cs:21]: Use `nameof` here?

**Comment by kundadebdatta** [comment]
The [original PR](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4379) to master has been merged. Therefore, closing this PR.

**Pull request closed without merging**